tests/python/pants_test/pants_run_integration_test.py

```diff
@@ -110,6 +110,12 @@ def hermetic_env_whitelist(cls):
     return [
         # Used in the wrapper script to locate a rust install.
         'HOME',
+
+        # Used in the wrapper script to locate a python 2.7 binary via `which`; also used in
+        # pants.pex for the same purpose via a pex header shebang of `#!/usr/bin/env python2.7`.
+        'PATH',
```

Comment:
This feels like a bit of a can of worms...

Would it be possible to filter this down to only path entries containing `python` or something? This (in theory, if it were working) should be a good defense against the bad practice of having PATH dependencies that are not fetched with binutils.

Reply:
> All this said: any ideas about how the heck this has ever worked on CI or any machine for that matter since hermetic was introduced?! I'm stumped.

Heh. @cosmicexplorer not that you're responsible for this in the slightest, but since you've been working on PATH filtering elsewhere: any ideas?

Reply:
Looking into it now. Anecdotally, `unset PATH; /usr/bin/env python2.7` seems to work on my linux machine though.

Reply:
(I have no clue why this would happen, reading up on `env` next)

Reply:
The immediate issue is not `env` but `which` fwiw: https://github.com/pantsbuild/pants/blob/master/pants#L50

Reply:
The only thing I can think of is that `which` might be implemented as a shell builtin? But upon searching I cannot find a single shell that does that.

Reply:
...and immediately afterward I find that zsh does that by default so I wouldn't be surprised if some others did too (although bash doesn't on this arch linux machine).

Reply:
And so, to the heart of the matter, I assume my examples fail for you as well.